test/com/amazon/ion/impl/bin/IonManagedBinaryWriterTest.java

```diff
@@ -179,6 +159,21 @@ public void testSystemSymbol() throws Exception
         assertValue("name");
     }
 
+    @Test
+    public void testLocalSymbolTableAppend() throws Exception
+    {
+        writer.writeSymbol("taco");
+        writer.flush();
+        writer.writeSymbol("burrito");
+        writer.finish();
+        IonReader reader = system().newReader(writer.getBytes());
+        reader.next();
+        assertEquals(reader.getSymbolTable().findSymbol("taco"), 15);
+        assertEquals(reader.getSymbolTable().findSymbol("burrito"), -1);
+        reader.next();
+        assertEquals(reader.getSymbolTable().findSymbol("burrito"), 16);
```

Comment:
Also check to make sure it "taco" still has symbol ID 15.

Reply:
done